opm/simulators/linalg/gpuistl/detail/preconditionerKernels/ILU0Kernels.cu

```diff
@@ -120,17 +120,21 @@ namespace
         }
     }
 
-    template <class T, int blocksize>
-    __global__ void cuLUFactorizationSplit(T* reorderedLowerMat,
+    template <int blocksize, class InputScalar, class OutputScalar>
+    __global__ void cuLUFactorizationSplit(InputScalar* srcReorderedLowerMat,
                                            int* lowerRowIndices,
                                            int* lowerColIndices,
-                                           T* reorderedUpperMat,
+                                           InputScalar* srcReorderedUpperMat,
                                            int* upperRowIndices,
                                            int* upperColIndices,
-                                           T* diagonal,
+                                           InputScalar* srcDiagonal,
```

Comment:
`srcDiagonal` can now probably be `const`?

Reply:
The srcMatrices are written to if we have copyResultToOtherMatrix enabled, which we enable in the mixed precision case. The naming is maybe a misleading. When incorporating an additional boolean into this function to split these two cases I still dont think we can have a const here only in one case, will this spoil compiler optimizations?